### DISCUSSION TITLE
GitHub Action on Pull Request only for certain branches

### DISCUSSION THREAD

**Discussion by danthedaniel** [community/community#26795] [Actions]
<p>I’m using Actions to set up Heroku+Vercel review apps for pull requests. I also use Git Flow, so release candidates and hot fixes are made as pull requests and get deployed to staging. I don’t want my actions related to review apps to run for RCs and hot fixes. I’ve added a couple rules to the workflow YAML configs:</p>
<pre><code class="lang-yaml">name: Set up review app

on:
  pull_request:
    types: [opened, reopened]
    branches-ignore:
      - 'rc/**'
      - 'hotfix/**'
</code></pre>
<p>It does not seem that the <code>branches-ignore</code> key is recognized. I’ve also tried <code>branches:</code> with negated patterns, ex:</p>
<pre><code class="lang-yaml">branches:
  - '**'
  - '!rc/**'
</code></pre>
<p>The docs that refer to the <code>branches</code> and <code>branches-ignore</code> keys always seem to mention them in the context of the <code>push</code> event. Is there any way to get this functionality for <code>pull_request</code> events besides adding a step in the action that checks the branch patterns? That would be confusing as the action would still show up on an RC’s pull request. That would only be acceptable if there was some way to indicate that the action was skipped.</p>

**Comment by airtower-luna** [comment, marked as answer]
<p>The branch filter for <code>pull_request</code> events works based on the name of the <em>target</em> (“base”) branch, not the branch holding the changes you want to merge in. So those filters you’ve tried will skip the workflow if the PR is trying to merge into an RC or hotfix branch. See <a href="https://docs.github.com/en/actions/learn-github-actions/workflow-syntax-for-github-actions#onpushpull_requestbranchestags">documentation on <code>on.&lt;push|pull_request&gt;.&lt;branches|tags&gt;</code></a>:</p>
<blockquote>
<p>For a <code>pull_request</code> event, only branches and tags on the base are evaluated.</p>
</blockquote>

**Reply by JJ** [reply]
This begs the question, is there a way to filter workflows based on the name of the branch with the actual changes?

**Reply by villelahdenvuo** [reply]
Is there a feature request somewhere where we can vote on it? It would be very useful to be able to e.g. ignore `dependabot/**` branches from linting etc.

**Reply by antony** [reply]
most pull requests are against main/master, so not having a way to filter on the PR branch itself is extremely limiting

**Reply by oerp-odoo** [reply]
I have situation where I need to run workflows on automatically created PRs. But they seem to not run on pull_request event initially (only if later you update that PR). So I was thinking of excluding such branches (as they have naming pattern that you can match) from pull_request event and only use push event. But if you enable pull_request event, then all pull requests will be matched..

If I don't exclude it, then later 2 identical workflows will be run. One for push and another for pull_request, which is annoying and wastes runners time.

**Reply by LucAwater** [reply]
Same here, I'm using release-please to create release PRs, but I don't want to run all the checks on it that I'm running on other (feature) PRs. Right now I'm using path filtering on files like CHANGELOG.md and the release please manifest, but it feels hacky and is kind of invisible for devs.

**Reply by publicJorn** [reply]
```
To run a job based on the pull request's head branch name (as opposed to the pull request's base branch name), use the github.head_ref context in a conditional.
```
https://docs.github.com/en/actions/writing-workflows/choosing-when-your-workflow-runs/events-that-trigger-workflows#running-your-pull_request-workflow-based-on-the-head-or-base-branch-of-a-pull-request

**Reply by shanti-razr** [reply]
using this method makes logs very difficult to parse as every pull request will create a skipped run and PR checks more annoying to read as there are a lot of skipped runs on all new PRs. Is it possible to not have skipped runs by targeting a change branch, and just have the workflow not run at all?
